### PR TITLE
Fix #6890: add filter options in db-panel

### LINKED ISSUE #6890: [ENH] Add filter in debug panel
add filter types in db panel with type querys execute.

For example :  

screenshot_1

### PR DESCRIPTION
**Pull request by pana1990**

### CODE REVIEW

File: extensions/debug/panels/DbPanel.php [hunk 1]
```diff
@@ -177,7 +182,12 @@ protected function getQueryType($timing)
         $timing = ltrim($timing);
         preg_match('/^([a-zA-z]*)/', $timing, $matches);
 
-        return count($matches) ? $matches[0] : '';
+        if (count($matches)) {
```
Comment (githubjeka) [inline extensions/debug/panels/DbPanel.php]: This method is **getQueryType** and you shouldn't **set** `type` in here.

File: extensions/debug/panels/DbPanel.php [hunk 2]
```diff
@@ -190,4 +200,15 @@ public function isQueryCountCritical($count)
...
+
+    /**
+     * Returns array query types
+     *
+     * @return array
+     * @since 2.0.3
+     */
+    public function getTypes()
+    {
+        return $this->_types;
```
Comment (githubjeka) [inline extensions/debug/panels/DbPanel.php]: Better use 
```php
public function getTypes()
    {
        return
            array_reduce(
                $this->_models,
                function ($result, $item) {
                    $result[$item['type']] = $item['type'];
                    return $result;
                },
                array()
            );
    }
```

Comment (pana1990) [inline extensions/debug/panels/DbPanel.php]: of course, thanks!

**Comment by samdark** [comment]
Is it handling paginated results properly?

**Comment by pana1990** [comment]
I do not understand, What do you mean by that?

**Comment by githubjeka** [comment]
Does It work if pagination on?

**Comment by samdark** [comment]
Should actually be OK. All models are there at once.

**Comment by samdark** [comment]
Merged https://github.com/yiisoft/yii2/commit/b3850a4e73cff3cf2f7d8e9fb77bbdd933eda3e6

Thanks!

**Pull request closed without merging**